### PR TITLE
kakoune.cr: init at unstable-2021-11-12

### PR DESCRIPTION
**Pull request by malte-v**
Co-authored-by: Sebastian Zivota <sebastian.zivota@mailbox.org>

This PR continues the work of @loewenheim on #115011. I've updated the program to the latest revision and removed the dependencies on `bat`, `fd` and `fzf`, since these are strictly optional and not at all required for `kakoune.cr`'s functionality.

Additionally, I've moved the files to the location in the source tree where `kak-lsp` already is, since `kakoune.cr` is not an editor unto itself but rather an add-on tool for Kakoune, just like `kak-lsp`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/misc/kakoune-cr/default.nix [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  src = fetchFromGitHub {
+    repo = "kakoune.cr";
+    owner = "alexherbo2";
+    rev = "43d4276e1d173839f335ff60f205b89705892e00";
+    hash = "sha256-xFrxbnZl/49vGKdkESPa6LpK0ckq4Jv5GNLL/G0qA1w=";
+  };
+
+  propagatedUserEnvPkgs = [ jq ];
+
+  crystalBinaries.kcr.src = "src/main.cr";
```
Comment (peterhoeg) [inline pkgs/tools/misc/kakoune-cr/default.nix:21]: You don't need this with `format = "shards";`

File: pkgs/tools/misc/kakoune-cr/default.nix [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    hash = "sha256-JxhIEmjiGrisaarA1sX1AfzNjHNIm9xjyPs/nG1uL/U=";
+  };
+in
+crystal.buildCrystalPackage rec {
+  pname = "kakoune.cr";
+  version = "unstable-2021-11-12";
+
+  src = fetchFromGitHub {
+    repo = "kakoune.cr";
+    owner = "alexherbo2";
```
Comment (peterhoeg) [inline pkgs/tools/misc/kakoune-cr/default.nix:14]: By convention, owner comes before repo.

File: pkgs/tools/misc/kakoune-cr/default.nix [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+  '';
+
+  postInstall = ''
+    install -Dm555 share/kcr/commands/*/kcr-* -t $out/bin
+    install -Dm444 share/kcr/applications/kcr.desktop -t $out/share/applications
+    install -Dm444 ${icon} $out/share/icons/hicolor/scalable/apps/kcr.svg
+    cp -r share/kcr $out/share/
+  '';
+
+  doInstallCheck = false;
```
Comment (peterhoeg) [inline pkgs/tools/misc/kakoune-cr/default.nix:40]: Why?

**Review by peterhoeg — CHANGES_REQUESTED** [review]
A few minor things need addressing

**Comment by loewenheim** [comment]
Thank you very much for taking care of this!

File: pkgs/tools/misc/kakoune-cr/default.nix (hunk 3 above)
Comment (malte-v) [reply pkgs/tools/misc/kakoune-cr/default.nix]: The default `installCheckPhase` just calls all binaries with `--help`. Most of the binaries in this package don't recognize this flag, so I disabled it initially. I've changed this to calling `--help` on `bin/kcr`, which recognizes the flag.

**Pull request merged**